pingcap/tidb
# planner: validate row expression value size for point get plan

### What problem does this PR solve?

Issue Number: close #19894 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:  
1. check parameters in  `In` statment

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test

before
```
tidb> create table t2(a int, b int, c int, primary key(a, b, c));
Query OK, 0 rows affected (0.02 sec)

tidb> select * from t2 where (a, b, c) in ((1, 1, 1, 1), (2, 2, 2));
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    1
Current database: test

ERROR 1105 (HY000): runtime error: index out of range [3] with length 3
```
after
```
tidb> create table t2(a int, b int, c int, primary key(a, b, c));
Query OK, 0 rows affected (0.02 sec)

tidb> select * from t2 where (a, b, c) in ((1, 1, 1, 1), (2, 2, 2));
ERROR 1241 (21000): Operand should contain 3 column(s)
```

Side effects

- none

### Release note 
- fix panic occurred when querying with for multiple columns
